core/src/main/java/org/frankframework/pipes/AbstractPipe.java

```diff
@@ -196,28 +199,36 @@ public void setLocker(Locker locker) {
 	@Override
 	public void registerForward(PipeForward forward) throws ConfigurationException {
 		String forwardName = forward.getName();
-		if(forwardName != null) {
-			PipeForward current = pipeForwards.get(forwardName);
-			if (current==null){
-				pipeForwards.put(forwardName, forward);
-			} else {
-				if (forward.getPath()!=null && forward.getPath().equals(current.getPath())) {
-					ConfigurationWarnings.add(this, log, "forward ["+forwardName+"] is already registered");
-				} else {
-					log.info("PipeForward [{}] already registered, pointing to [{}]. Ignoring new one, that points to [{}]", forwardName, current.getPath(), forward.getPath());
-				}
+
+		if (forwardName == null) {
+			throw new ConfigurationException("forward without a name");
+		}
+
+		final Set<Forward> allowedForwards = getAllowedForwards();
+		if (allowedForwards.stream().noneMatch(Forward::wildcard)) {
+			if (!allowedForwards.stream().map(Forward::name).toList().contains(forwardName)) {
+				ConfigurationWarnings.add(this, log, "The forward [" + forwardName + "] does not exist and cannot be used in this pipe");
```

Comment:
```suggestion
				throw new ConfigurationException("The forward [" + forwardName + "] does not exist and cannot be used in this pipe");
```

Reply:
Ik wou dit oorspronkelijk ook doen, omdat een taal met strenge regels een goed plan is, maar dit heeft wel een groot nadeel. Als wij een wildcard forward vergeten toe te voegen (zoals bij `XmlIf` het geval was) werkt die pipe niet met wildcard forwards, omdat er een exceptie gegooid wordt. Een warning is inderdaad iets minder in your face, maar heeft dit probleem niet.

Reply:
Ja, maar toch is het handig dit te doen, wij maken immers nooit fouten. 😄